Automattic/simplenote-android
# Add Add Tag

### Fix
Add a floating action button to the **_Edit Tags_** screen to add a tag.  See the animation below for illustration.

Fast|Slow
-|-
<kbd><a href="https://user-images.githubusercontent.com/3827611/100007866-ef178e80-2d89-11eb-9d2a-b6c10a77e814.gif"><img src="https://user-images.githubusercontent.com/3827611/100007866-ef178e80-2d89-11eb-9d2a-b6c10a77e814.gif" width="300"></a></kbd>|<kbd><a href="https://user-images.githubusercontent.com/3827611/100007851-e6bf5380-2d89-11eb-9189-95a358c3caa0.gif"><img src="https://user-images.githubusercontent.com/3827611/100007851-e6bf5380-2d89-11eb-9189-95a358c3caa0.gif" width="300"></a></kbd>

When a valid tag name is input and the **_Save_** button is tapped in the **_Add Tag_** dialog, the added tag is shown in the list on the **_Edit Tags_** screen.  See the screenshots below for illustration.

![add_add_tag](https://user-images.githubusercontent.com/3827611/100003723-ca201d00-2d83-11eb-916c-7846230895d9.png)

The **_Add Tag_** dialog will show an error message and disable the **_Save_** button if the tag is empty, contains spaces, already exists, or is too long.  See the screenshots below for illustration.

![add_add_tag_error](https://user-images.githubusercontent.com/3827611/100003658-b1b00280-2d83-11eb-8027-6bbf4084f7fd.png)

### Test
1. Open navigation drawer.
2. Tap ***Edit*** button in ***Tags*** header.
3. Notice floating action button in bottom, right/end corner or ***Edit Tags*** screen.
4. Long-press floating action button.
5. Notice ***Add Tag*** message is shown.
6. Tap floating action button.
7. Notice ***Add Tag*** dialog is shown.
8. Enter space character in ***Tag name*** field.
9. Notice ***Tags cannot contain spaces*** error message is shown.
10. Notice ***Save*** button is disabled.
11. Delete space character from ***Tag name*** field.
12. Notice ***Tags cannot be empty*** error message is shown.
13. Notice ***Save*** button is disabled.
14. Enter any character in ***Tag name*** field.
15. Notice ***Save*** button is enabled.
16. Enter existing tag name in ***Tag name*** field.
17. Notice ***Tag already exists*** error message is shown.
18. Notice ***Save*** button is disabled.
19. Enter four England flag emojis (i.e. `🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿🏴󠁧󠁢󠁥󠁮󠁧󠁿`) in ***Tag name*** field.
20. Notice ***Tag length too long*** error message is shown.
21. Notice ***Save*** button is disabled.
22. Enter non-existing tag name in ***Tag name*** field.
23. Notice ***Save*** button is enabled.
24. Tap ***Save*** button.
25. Notice tag from Step  is added to list.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

#### Note
Contact me for an installable build.

### Release
These changes do not require release notes.
